### ISSUE TITLE
Including the branch name in the version?

### ISSUE THREAD

**Issue by lokkju** [dwijnand/sbt-dynver#51]
Any chance of a simple option to enable including branch names other than master in the version string?

**Comment by dwijnand** [comment]
Is there much demand for this support out the box?

Without out the box support you can do this by customising the version string, as defined in https://github.com/dwijnand/sbt-dynver#custom-version-string.

**Comment by lokkju** [comment]
I think it's a pretty common use case, and standard in many build systems; it's could be included as part of build metadata with semver, at the end.

More bringing it up a discussion point; it'd be nice to see the branch name in the metadata, but after thinking about it, it's not required for version uniqueness.

**Comment by raboof** [comment]
I don't think it would be particularly useful to include this in the default version string, but it would be neat to offer a utility function to easily get the git branch for use in your custom version string

**Comment by dwijnand** [comment]
A utility function leads to utility function variants, overloads, default params, etc.  I'd rather not.  Do something like

```scala
scala> scala.util.Try(scala.sys.process.Process("git rev-parse --symbolic-full-name --abbrev-ref HEAD").!!).toOption.map(_.dropRight(1))
res0: Option[String] = Some(master)
```

**Comment by atais** [comment]
I had a similar need and I do not think the lib is missing anything. It's actually all there.

Here is my very simplified auto-versioning.

```
.settings(
    version := myVersion(dynverGitDescribeOutput.value),
    dynver ~= (_.replace('+', '-'))
)

def myVersion(out: Option[sbtdynver.GitDescribeOutput]): String = {
  val branchName = Process("git rev-parse --symbolic-full-name --abbrev-ref HEAD").!!.dropRight(1)
  out match {
    case Some(v) if branchName == "develop" =>
      s"${v.commitSuffix.distance}-${v.ref.value}"
    case Some(v) =>
      s"${v.commitSuffix.distance}-${v.ref.value}-${branchName}"
    case None =>
      "0-undefined"
  }
}
```
